services/src/services/authentication/authentication.hooks.js

```diff
@@ -0,0 +1,93 @@
+const errors      = require('@feathersjs/errors');
+const ecc         = require('elliptic');
+const logger      = require('winston');
+const dbtimestamp = require('../../hooks/dbtimestamp');
+
+module.exports = {
+  before: {
+    all: [],
+    find: [],
+    get: [],
+    create: [
+      /* Ensure that the request is valid
+       */
+      (hook) => {
+        if ((!hook.data.uuid) ||
+          (!hook.data.signature)) {
+          throw new errors.BadRequest('Missing uuid and/or signature from request');
+        }
+      },
+
+      /* Ensure that the uuid actually exists in the registration service
+       */
+      (hook) => {
+        const reg = hook.app.service('registration');
+        return reg.find({ query: { uuid: hook.data.uuid }}).then(records => {
+          if (records.total < 1) {
+            throw new errors.NotFound('Unknown instance');
+          }
+          else {
+            /*
+             * Since we've already looked up the UUID, let's pass it along for
+             * future hook use
+             */
+            hook.data.record = records.data[0];
+          }
+        }).catch((err) => {
+          /* Expected that malformed uuids will throw a
+           * `SequelizeDatabaseError`
+           */
+          logger.error('Authentication.create()', err);
+          throw new errors.NotFound('Unknown instance');
+        });
+      },
+
+      /*
+       * Using the public key passed by the previous hook, ensure that the
+       * signature of the UUID matches appropriately
+       */
+      (hook) => {
+        let record = hook.data.record;
+        let ec = new ecc.ec(record.curve);
+        let key = ec.keyFromPublic(record.pubKey, 'hex');
+
+        let invalidError = new errors.BadRequest('Invalid signature provided');
```

Comment:
NIT (mostly for my understanding really): I don't measure well the impact of creating this each time, but IIUC this is going to be always run, when it's supposed to be useful/used when something goes wrong. I hope/suppose the call is not too costly. As it's the service side, I guess this is not going to be an issue now, but could become once called often (?)

Reply:
There is no significant cost overhead as far as I am aware with constructing "error" objects, which as far as I am aware are nothing special compared to other "objects" in JavaScript.

I know Java has overhead related to Exception construction, but to my understanding JavaScript has no similar performance penalties